arch/rv32i/src/lib.rs

```diff
@@ -81,25 +86,27 @@ pub unsafe fn init_memory() {
     tock_rt0::zero_bss(&mut _szero, &mut _ezero);
 }
 
-/// Tell the MCU what address the trap handler is located at. The trap handler
-/// is called on exceptions and for interrupts.
+pub enum PermissionMode {
+    Machine = 0x1,
+    Supervisor = 0x10,
+    User = 0x100,
```

Comment:
Can we follow the RISC-V spec privileged order and keep the reserved privileged level? Ideally it would be best to use the encoding from the spec as well.
U -> 0
S -> 1
Reserved -> 2
M -> 3